core/services/relay/evm/evm.go

```diff
@@ -931,16 +933,52 @@ func generateTransmitterFrom(ctx context.Context, rargs commontypes.RelayArgs, e
 			relayConfig.DualTransmissionConfig,
 		)
 	case commontypes.CCIPExecution:
-		transmitter, err = cciptransmitter.NewTransmitterWithStatusChecker(
-			configWatcher.chain.TxManager(),
-			fromAddresses,
-			gasLimit,
-			effectiveTransmitterAddress,
-			strategy,
-			checker,
-			configWatcher.chain.ID(),
-			ethKeystore,
-		)
+		if configWatcher.chain.Config().EVM().ChainType() == chaintype.ChainTron {
+			transmitter, err = cciptransmitter.NewTronTransmitterWithStatusChecker(
```

Comment:
How about we try injecting the Tron write side as `ocrtypes.ContractTransmitter` type, which is chain agnostic and on the layer above. Here you're reusing the EVM write side and using the `evm.Transmitter` type which is going to be harder to fulfil imo

I'd hook into and change the `newOnChainContractTransmitter` fn - in case of Tron, we'd try to instantiate this existing Tron `ocrtypes.ContractTransmitter`: https://github.com/smartcontractkit/chainlink-internal-integrations/blob/develop/tron/relayer/ocr2/contract_transmitter.go#L32

Reply:
This would require us importing the Tron codebase here (currently internal).